website_sale_product_pack/models/product_template.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    def write(self, vals):
+        """ For keep the consistent and prevent bugs within the e-commerce,
+         we force that all childs of a parent pack
+         stay publish when the parent is published.
+         Also if any of the childs of the parent pack became unpublish,
+         we unpublish the parent."""
+        if len(vals) == 1 and "website_published" in vals and vals["website_published"]:
```

Comment:
Why `len(vals) == 1`

Reply:
because if came more values i don't want to change for all the records(parent childs) the other values. This way i ensure that only change the `website published` field not other fields.

Are you agree with that or have another idea?

Reply:
@chienandalu

Reply:
For the children case, I'd just throw a warning. For the parents case, it'd be ok. You can update the parents anyway so you don't need to restrict the length of the vals. Otherwise you could be missing some cases.

Reply:
I've just change the approach of this. 
When publish a pack, we publish all of the child's also.
When unpublish one of the child's, we unpublish the parent product

Reply:
> When publish a pack, we publish all of the child's also.

I think that approach is problematic and could create a chain of conflicts when those pack components are in several packs with different publishing states. We should just throw a warning to avoid publishing such pack and let the user review the state.

> When unpublish one of the child's, we unpublish the parent product

Not as conflicting, but the user could not be aware of such behavior. Maybe the warning could be the way to go again.

Reply:
Yes Iagree with you, i've made the changes in the commit.